README.v4.md

```diff
@@ -0,0 +1,526 @@
+
+### toValidationError
+
+A curried version of `fromZodError` meant to be used for FP (Functional Programming). Note it first takes the options object if needed and returns a function that converts the `zodError` to a `ValidationError` object
+
+```js
+toValidationError(options) => (zodError) => ValidationError
+```
+
+#### Example using fp-ts
+
+```typescript
+import * as Either from 'fp-ts/Either';
+import { z as zod } from 'zod/v4';
+import { toValidationError, ValidationError } from 'zod-validation-error/v4';
+
+// create zod schema
+const zodSchema = zod
+  .object({
+    id: zod.int().positive(),
+    email: zod.email(),
+  })
+  .brand<'User'>();
+
+export type User = zod.infer<typeof zodSchema>;
+
+export function parse(
+  value: zod.input<typeof zodSchema>
+): Either.Either<ValidationError, User> {
+  return Either.tryCatch(() => schema.parse(value), toValidationError());
+}
+```
+
+## FAQ
+
+### How to distinguish between errors
+
+Use the `isValidationErrorLike` type guard.
+
+#### Example
+
+Scenario: Distinguish between `ValidationError` VS generic `Error` in order to respond with 400 VS 500 HTTP status code respectively.
+
+```typescript
+import * as Either from 'fp-ts/Either';
+import { z as zod } from 'zod/v4';
+import { isValidationErrorLike } from 'zod-validation-error/v4';
+
+try {
+  func(); // throws Error - or - ValidationError
+} catch (err) {
+  if (isValidationErrorLike(err)) {
+    return 400; // Bad Data (this is a client error)
+  }
+
+  return 500; // Server Error
+}
+```
+
+### How to use `ValidationError` outside `zod`
+
+It's possible to implement custom validation logic outside `zod` and throw a `ValidationError`.
+
+#### Example 1: passing custom message
+
+```typescript
+import { ValidationError } from 'zod-validation-error/v4';
+import { Buffer } from 'node:buffer';
+
+function parseBuffer(buf: unknown): Buffer {
+  if (!Buffer.isBuffer(buf)) {
+    throw new ValidationError('Invalid argument; expected buffer');
+  }
+
+  return buf;
+}
+```
+
+#### Example 2: passing custom message and original error as cause
+
+```typescript
+import { ValidationError } from 'zod-validation-error/v4';
+
+try {
+  // do something that throws an error
+} catch (err) {
+  throw new ValidationError('Something went deeply wrong', { cause: err });
+}
+```
+
+### How to use `ValidationError` with custom "error map"
+
+Zod supports customizing error messages by providing a custom "error map". You may combine this with `zod-validation-error` to produce user-friendly messages.
+
+#### Example 1: produce user-friendly error messages using the `errorMap` property
+
+If all you need is to produce user-friendly error messages you may use the `errorMap` property.
+
+```typescript
+import { z as zod } from 'zod/v4';
+import { crea } from 'zod-validation-error/v4';
+
+zod.setErrorMap(errorMap);
```

Comment:
question: is it still the case for v4, or should we update it to use the `zod.config` API?

Reply:
bloody hell